### PR TITLE
Use scheduling create_task util to create tasks

### LINKED ISSUE #1807: Replace bot.loop.create_task usages with scheduling.create_task
Since 0b726398d2135ab414374412fa85f791569e3640 we accept a loop kwarg to `scheduling.create_task` which can be used to create a task with a not yet running loop where `asyncio.create_task` doesn't work directly.

Because the created tasks currently don't have any error handling they can hide errors in development until the task object is destroyed (if that occurs at all) which logs the exception. The scheduling alternative attaches a callback which logs exceptions to prevent this.

For every task the call should look something like this:
```py
self.placeholder_task = scheduling.create_task(coro, event_loop=<the loop previously used to create the task>)
```

### PR DESCRIPTION
**Pull request by Numerlor**
closes #1807 
replaces `loop.create_task` and `asyncio.create_task` usages with the scheduling util so exception logging is attached to created tasks

### CODE REVIEW

File: bot/exts/help_channels/_cog.py [hunk 1]
```diff
@@ -507,7 +507,7 @@ async def wait_for_dormant_channel(self) -> discord.TextChannel:
         """Wait for a dormant channel to become available in the queue and return it."""
         log.trace("Waiting for a dormant channel.")
 
-        task = asyncio.create_task(self.channel_queue.get())
+        task = scheduling.create_task(self.channel_queue.get())
```
Comment (ChrisLovering) [inline bot/exts/help_channels/_cog.py:510]: Any reason not to use the bot loop here?

File: bot/exts/moderation/defcon.py [hunk 2]
```diff
@@ -205,7 +205,7 @@ def _update_channel_topic(self) -> None:
         new_topic = f"{BASE_CHANNEL_TOPIC}\n(Threshold: {humanize_delta(self.threshold) if self.threshold else '-'})"
 
         self.mod_log.ignore(Event.guild_channel_update, Channels.defcon)
-        asyncio.create_task(self.channel.edit(topic=new_topic))
+        scheduling.create_task(self.channel.edit(topic=new_topic))
```
Comment (ChrisLovering) [inline bot/exts/moderation/defcon.py:208]: Any reason not to use the bot loop here?

File: bot/exts/utils/reminders.py [hunk 3]
```diff
@@ -80,7 +80,7 @@ def ensure_valid_reminder(self, reminder: dict) -> t.Tuple[bool, discord.User, d
                 f"Reminder {reminder['id']} invalid: "
                 f"User {reminder['author']}={user}, Channel {reminder['channel_id']}={channel}."
             )
-            asyncio.create_task(self.bot.api_client.delete(f"bot/reminders/{reminder['id']}"))
+            scheduling.create_task(self.bot.api_client.delete(f"bot/reminders/{reminder['id']}"))
```
Comment (ChrisLovering) [inline bot/exts/utils/reminders.py:83]: Any reason not to use the bot loop here?

**Review by ChrisLovering — APPROVED** [review]

File: bot/exts/help_channels/_cog.py (hunk 1 above)
Comment (Numerlor) [reply bot/exts/help_channels/_cog.py:510]: If the loop is not passed in, the util itself uses asyncio.create_task which gets the running loop

**Review by Xithrius — APPROVED** [review]

**Pull request merged**